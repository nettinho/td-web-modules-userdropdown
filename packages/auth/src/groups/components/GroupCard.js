import React from "react";
import PropTypes from "prop-types";
import pathToRegexp from "path-to-regexp";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Button, Card, Icon } from "semantic-ui-react";
import { Link } from "react-router-dom";
import { FormattedMessage } from "react-intl";
import { ConfirmModal } from "@truedat/core/components";
import { deleteGroup } from "../routines";
import routes from "../routes";

const toEditPath = pathToRegexp.compile(routes.GROUP_EDIT);

export const GroupCard = ({
  group: { name, description, id },
  deleteGroup
}) => (
  <Card key={id}>
    <Card.Content>
      <Card.Header>
        <Icon name="group" /> {name}
      </Card.Header>
      <Card.Description>{description}</Card.Description>
    </Card.Content>
    <Card.Content extra>
      <div className="ui two buttons">
        <Button
          basic
          color="green"
          as={Link}
          to={toEditPath({ id })}
          content={<FormattedMessage id="actions.edit" />}
        />
        <ConfirmModal
          icon="trash"
          trigger={
            <Button
              basic
              negative
              content={<FormattedMessage id="actions.delete" />}
            />
          }
          header={
            <FormattedMessage id="group.actions.delete.confirmation.header" />
          }
          content={
            <FormattedMessage
              id="group.actions.delete.confirmation.content"
              values={{ name: <i>{name}</i> }}
            />
          }
          handleSubmit={() => deleteGroup({ id })}
        />
      </div>
    </Card.Content>
  </Card>
);

GroupCard.propTypes = {
  group: PropTypes.object,
  deleteGroup: PropTypes.func
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ deleteGroup }, dispatch)
});

export default connect(
  null,
  mapDispatchToProps
)(GroupCard);
