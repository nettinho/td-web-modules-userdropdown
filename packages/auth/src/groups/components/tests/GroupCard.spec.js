import React from "react";
import { shallow } from "enzyme";
import { GroupCard } from "../GroupCard";

describe("<GroupCard />", () => {
  const deleteGroup = jest.fn();
  const group = {
    id: 1,
    name: "administrators",
    description: "Administrators"
  };
  const props = { group, deleteGroup };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<GroupCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
