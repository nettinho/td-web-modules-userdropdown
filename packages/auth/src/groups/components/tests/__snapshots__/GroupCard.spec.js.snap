// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupCard /> matches the latest snapshot 1`] = `
<Card
  key="1"
>
  <CardContent>
    <CardHeader>
      <Icon
        as="i"
        name="group"
      />
       
      administrators
    </CardHeader>
    <CardDescription>
      Administrators
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <div
      className="ui two buttons"
    >
      <Button
        as={[Function]}
        basic={true}
        color="green"
        content={
          <FormattedMessage
            id="actions.edit"
            values={Object {}}
          />
        }
        role="button"
        to="/groups/1/edit"
      />
      <ConfirmModal
        content={
          <FormattedMessage
            id="group.actions.delete.confirmation.content"
            values={
              Object {
                "name": <i>
                  administrators
                </i>,
              }
            }
          />
        }
        handleSubmit={[Function]}
        header={
          <FormattedMessage
            id="group.actions.delete.confirmation.header"
            values={Object {}}
          />
        }
        icon="trash"
        trigger={
          <Button
            as="button"
            basic={true}
            content={
              <FormattedMessage
                id="actions.delete"
                values={Object {}}
              />
            }
            negative={true}
            role="button"
          />
        }
      />
    </div>
  </CardContent>
</Card>
`;
